gensim/test/test_scripts.py

```diff
@@ -1,108 +1,186 @@
+        segment_and_write_all_articles(self.fname, tmpf, workers=1)
+
+        expected_num_articles = 106
+        num_articles = sum(1 for line in smart_open(tmpf))
+        self.assertEqual(num_articles, expected_num_articles)
+
+    def test_segment_and_write_all_articles(self):
+        tmpf = get_tmpfile('script.tst.json')
+        segment_and_write_all_articles(self.fname, tmpf, workers=1, include_interlinks=True)
+
+        # Get the first line from the text file we created.
+        with open(tmpf) as f:
+            first = next(f)
+
+        # decode JSON line into a Python dictionary object
+        article = json.loads(first)
+        title, section_titles, interlinks = article['title'], article['section_titles'], article['interlinks']
+
+        self.assertEqual(title, self.expected_title)
+        self.assertEqual(section_titles, self.expected_section_titles)
+
+        # Check interlinks
+        self.assertTrue(interlinks['self-governance'] == 'self-governed')
+        self.assertTrue(interlinks['Hierarchy'] == 'hierarchical')
+        self.assertTrue(interlinks['Pierre-Joseph Proudhon'] == 'Proudhon')
+
+
+class TestWord2Vec2Tensor(unittest.TestCase):
+    def setUp(self):
+        self.datapath = datapath('word2vec_pre_kv_c')
+        self.output_folder = get_tmpfile('')
+        self.metadata_file = self.output_folder + '_metadata.tsv'
+        self.tensor_file = self.output_folder + '_tensor.tsv'
+        self.vector_file = self.output_folder + '_vector.tsv'
+
+    def testConversion(self):
+        word2vec2tensor(word2vec_model_path=self.datapath, tensor_filename=self.output_folder)
+
+        try:
+            with smart_open(self.metadata_file, 'rb') as f:
+                metadata = f.readlines()
+        except Exception:
+            if not os.path.isfile(os.path.join(self.metadata_file)):
+                self.fail(
+                    'Metadata file %s creation failed. \
+                    Check the parameters and input file format.' % self.metadata_file
+                    )
+        try:
+            with smart_open(self.tensor_file, 'rb') as f:
+                vectors = f.readlines()
+        except Exception:
+            if not os.path.isfile(os.path.join(self.tensor_file)):
+                self.fail(
+                    'Tensor file %s creation failed. \
+                    Check the parameters and input file format.' % self.tensor_file
+                    )
+
+        # check if number of words and vector size in tensor file line up with word2vec
+        with smart_open(self.datapath, 'rb') as f:
+            first_line = f.readline().strip()
+
+        number_words, vector_size = map(int, first_line.split(b' '))
+        if not len(metadata) == len(vectors) == number_words:
+            self.fail(
+                'Metadata file %s and tensor file %s \
+                imply different number of rows.' % (self.metadata_file, self.tensor_file)
+                )
+
+        # write word2vec to file
```

Comment:
The strange part of the test, you copy part of the code from script to test this script, why?

Reply:
I'm, sorry, could you be a bit more specific? Do you mean the lines from 154-157? If so, the idea was to take the metadata and tensor components that were separated and put the back together in w2v style without any gensim functions and make sure they are the same as what is created by the `word2vec2tensor.py` script. I guess it is basically just tested `to_utf8` though. Is it fine to remove it?

Reply:
I think it's fine to remove it, yes